AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java

```diff
@@ -433,57 +435,56 @@ private void configureToolbarInternal(boolean recur) {
         }
     }
 
-
-    @SuppressWarnings("deprecation") // Tracked as https://github.com/ankidroid/Anki-Android/issues/8293
-    @Override
-    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
-        super.onActivityResult(requestCode, resultCode, intent);
-        Timber.d("onActivityResult (requestCode = %d, resultCode = %d)", requestCode, resultCode);
-
-        // rebuild action bar
+    ActivityResultLauncher<Intent> onRequestPreviewActivityResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
         configureToolbar();
-
-        // boot back to deck picker if there was an error
-        if (resultCode == DeckPicker.RESULT_DB_ERROR || resultCode == DeckPicker.RESULT_MEDIA_EJECTED) {
-            closeStudyOptions(resultCode);
+        if (result.getResultCode() == DeckPicker.RESULT_DB_ERROR || result.getResultCode() == DeckPicker.RESULT_MEDIA_EJECTED) {
```

Comment:
I believe our coding standard go for "explicit parantheses" - by which I mean, even though they are not strictly necessary we put parentheses around all boolean logic clauses just to help tired brains read the if statements without thinking more than necessary during code reviews ;-)

```suggestion
        if ((result.getResultCode() == DeckPicker.RESULT_DB_ERROR) || (result.getResultCode() == DeckPicker.RESULT_MEDIA_EJECTED)) {
```